tsa-test/src/test/kotlin/org/ton/examples/checkers/Statements.kt

```diff
@@ -0,0 +1,202 @@
+package org.ton.examples.checkers
+
+import org.ton.test.utils.FIFT_STDLIB_RESOURCE
+import org.ton.test.utils.checkInvariants
+import org.ton.test.utils.extractResource
+import org.usvm.machine.TvmContext
+import org.usvm.machine.analyzeInterContract
+import org.usvm.machine.getFuncContract
+import org.usvm.test.resolver.TvmMethodFailure
+import org.usvm.test.resolver.TvmSuccessfulExecution
+import org.usvm.test.resolver.TvmTestInput.RecvInternalInput
+import kotlin.test.Test
+import kotlin.test.assertTrue
+
+class Statements {
+    private val targetContract = "/checkers/statements/target_contract.fc"
+    private val sendInRepeatChecker1 = "/checkers/statements/send_in_repeat_checker1.fc"
+    private val sendInRepeatChecker2 = "/checkers/statements/send_in_repeat_checker2.fc"
+    private val sendInRepeatChecker3 = "/checkers/statements/send_in_repeat_checker3.fc"
+    private val sendInRepeatChecker4 = "/checkers/statements/send_in_repeat_checker4.fc"
+    private val sendInWhileChecker1 = "/checkers/statements/send_in_while_checker1.fc"
+    private val sendInWhileChecker2 = "/checkers/statements/send_in_while_checker2.fc"
+    private val sendInWhileChecker3 = "/checkers/statements/send_in_while_checker3.fc"
+    private val sendInUntilChecker1 = "/checkers/statements/send_in_until_checker1.fc"
+    private val sendInUntilChecker2 = "/checkers/statements/send_in_until_checker2.fc"
+    private val sendInUntilChecker3 = "/checkers/statements/send_in_until_checker3.fc"
+    private val sendInUntilChecker4 = "/checkers/statements/send_in_until_checker4.fc"
+
+    private fun extractOpcode(cellData: String): Int {
+        val opCodeStr = cellData.slice(0..31)
+        return opCodeStr.toInt(2)
+    }
+
+    private fun checkFailedSendInStatement(checkerPathStr: String, possibleOpcodesList: List<List<Int>>) {
+        val analyzedPath = extractResource(targetContract)
+        val checkerPath = extractResource(checkerPathStr)
+
+        val checkerContract = getFuncContract(
+            checkerPath,
+            FIFT_STDLIB_RESOURCE,
+            isTSAChecker = true
+        )
+        val analyzedContract = getFuncContract(analyzedPath, FIFT_STDLIB_RESOURCE)
+
+        val tests = analyzeInterContract(
+            listOf(checkerContract, analyzedContract),
+            startContractId = 0,
+            methodId = TvmContext.RECEIVE_INTERNAL_ID
+        )
+
+        checkInvariants(
+            tests,
+            listOf { test ->
+                val result = test.result
+                // The result must be a failure with exit code 256
+                if (result is TvmMethodFailure) {
+                    assertTrue(result.exitCode == 256)
```

Comment:
Please, don't use asserts in predicates. Instead, you can write the following:
```
if (result.exitCode != 256) {
    return@listOf false
}
```

And if you use asserts out of lambdas, in such cases it is better to use `assertEquals` (the error message is more clear).